internal/service/models/exceptions/databaseError.go

```diff
@@ -0,0 +1,9 @@
+package exceptions
+
+type DatabaseError struct {
```

Comment:
Since we return raw errors in most cases this file can be removed.

Reply:
In Go it's not standard to have inheritance for exceptions like Java. For example in the repository we return the raw error.
https://github.com/wexinc/ps-tag-payment-scheduler/blob/a6a283834641399513ba31f31bac27fda8aa7279/internal/domain/payment/repo/payment_repo.go#L26-L34
IN the method that calls the repository we continue to return the raw error.
https://github.com/wexinc/ps-tag-payment-scheduler/blob/a6a283834641399513ba31f31bac27fda8aa7279/internal/domain/payment/service/payment_service.go#L67-L90
Then in the handler we deal with the fact that there is an error but we do not care about it's type.
https://github.com/wexinc/ps-tag-payment-scheduler/blob/a6a283834641399513ba31f31bac27fda8aa7279/internal/adapters/httpserver/payment_handler.go#L67

Reply:
I followed this tutorial for creating custom errors in Go https://www.digitalocean.com/community/tutorials/creating-custom-errors-in-go 
Here it explain how to create a custom error.
About the returning raw errors, it's not a best practice to rely in external errors, that's why we catch these and convert into something that we can control 100%. When the database, for example, return an error, we need to catch it and convert into something that our system can control its creation, instead of trust into something unknown.